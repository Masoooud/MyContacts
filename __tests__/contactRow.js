import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import ContactRow from '../src/components/contactRow';

describe('ContactRow test', () => {
  test('components rendered properly', () => {
    const contact = { name: { first: 'a', last: 'b' } };
    const navigation = jest.fn();

    const { getByTestId, getAllByTestId, queryByText, toJSON } = render(
      <ContactRow data={contact} navigation={navigation} />,
    );

    expect(getByTestId('button')).toBeTruthy();
    expect(getByTestId('first')).toBeTruthy();
    expect(getByTestId('last')).toBeTruthy();

    expect(getAllByTestId('button')).toHaveLength(1);
    expect(getAllByTestId('first')).toHaveLength(1);
    expect(getAllByTestId('last')).toHaveLength(1);

    expect(queryByText('a')).not.toBeNull();
    expect(queryByText('b')).not.toBeNull();

    expect(toJSON()).toMatchSnapshot();
  });
});
