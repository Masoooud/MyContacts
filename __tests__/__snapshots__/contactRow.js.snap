// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactRow test components rendered properly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 32,
    }
  }
  testID="button"
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 32,
        },
        Object {
          "fontFamily": "System",
          "fontWeight": "700",
        },
        Object {
          "marginRight": 5,
        },
      ]
    }
    testID="first"
  >
    a
  </Text>
  <Text
    style={
      Object {
        "fontSize": 32,
      }
    }
    testID="last"
  >
    b
  </Text>
</View>
`;
